### PR TITLE
fix(remote-read): return err instead of nil

### PR DESCRIPTION
**Pull request by Dentrax**
Signed-off-by: Furkan <furkan.turkal@trendyol.com>

#### What this PR does

Also adds missing `errcheck` linter since we already have a `errcheckexclude` file in the root.

Previous PR: https://github.com/grafana/cortex-tools/pull/237

```
$ go run cmd/mimirtool/main.go remote-read stats --address http://10.11.12.13:30090 --selector '{__name__!=""}'

INFO[0000] Created remote read client using endpoint 'http://10.11.12.13:30090/prometheus/api/v1/read'
INFO[0000] Querying time from=2022-09-27T08:48:54+03:00 to=2022-09-27T09:48:54+03:00 with selector={__name__!=""}
main: error: remote server http://10.11.12.13:30090/prometheus/api/v1/read returned HTTP status 404 Not Found: 404 page not found, try --help
exit status 1
```

#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/cortex-tools/issues/236

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

cc @aknuds1

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ linters:
     - gofmt
     - misspell
     - errorlint
+    - errcheck
```
Comment (aknuds1) [inline .golangci.yml:11]: `errcheck` is enabled by default, so there's no need to add it.

```suggestion
```

File: pkg/alertmanager/alertmanager.go [hunk 2]
```diff
@@ -356,7 +356,7 @@ func (am *Alertmanager) ApplyConfig(userID string, conf *config.Config, rawCfg s
 		return notifier
 	})
 	if err != nil {
-		return nil
+		return err
```
Comment (aknuds1) [inline pkg/alertmanager/alertmanager.go:359]: Please undo this change, since it diverges from the PR topic.
```suggestion
		return nil
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
At first pass, I see two changes that shouldn't be part of the PR.

**Comment by Dentrax** [comment]
MOVED: https://github.com/grafana/mimir/pull/3052

**Comment by aknuds1** [comment]
The Reopen button is disabled for me, saying the branch was force-pushed or recreated.

**Comment by Dentrax** [comment]
Oh, opened a new one: https://github.com/grafana/mimir/pull/3053
